src/@ionic-native/plugins/fcm/index.ts

```diff
@@ -0,0 +1,102 @@
+import { Plugin, IonicNativePlugin, Cordova } from '@ionic-native/core';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+export interface NotificationData {
+
+  /**
+   * Determines whether the notification was pressed or not
+   */
+
+  wasTapped: boolean;
+
+  /**
+   * Notification data hash item
+   */
+
+  [name: string]: any;
+
+}
+
+/**
+ * @name FCM
+ * @description
+ * Provides basic functionality for Firebase Cloud Messaging
+ *
+ * @usage
+ * ```
+ * import { FCM } from 'ionic-native';
+ *
+ * constructor(private fcm: FCM) { 
```

Comment:
The usage part is a bit to short. Can you describe the plugin a bit more? Orientate you by other plugins.
Don't put the plugin method call in to the constructor.